### PR TITLE
Feature/add rows cols

### PR DESCRIPTION
**Pull request by jramapuram**
Upstream has:
  - `rows()`
  - `cols()`

Adding these to the wrapper.

### CODE REVIEW

File: src/index.rs [hunk 1]
```diff
@@ -125,17 +125,39 @@ pub fn row(input: &Array, row_num: u64) -> Result<Array, AfError> {
...
 
+pub fn rows(input: &Array, first: u64, last: u64) -> Result<Array, AfError> {
+    let dims_err = input.dims();
+    let dims = match dims_err {
+        Ok(dim) =>  dim.clone(),
+        Err(e)  =>  panic!("Error unwrapping dims in rows(): {}", e),
+    };
+
+    index(input, &[Seq::new(first as f64, last as f64, 1.0)
+                    ,Seq::new(0.0, dims[1] as f64 - 1.0, 1.0)])
```
Comment (9prady9) [inline src/index.rs]: @jramapuram We do not need to retrieve dimensions, just doing &[Seq::new(first, last, 1), Seq::default()] would be enough for this. This way one less ffi call in terms of `input.dims()`.

`Seq::default` is the equivalent of `af_span` in C API of ArrayFire that spans all elements along that dimension.

File: src/index.rs [hunk 2]
```diff
@@ -125,17 +125,39 @@ pub fn row(input: &Array, row_num: u64) -> Result<Array, AfError> {
...
 }
 
+pub fn cols(input: &Array, first: u64, last: u64) -> Result<Array, AfError> {
+    let dims_err = input.dims();
+    let dims = match dims_err {
+        Ok(dim) =>  dim.clone(),
+        Err(e)  =>  panic!("Error unwrapping dims in cols(): {}", e),
+    };
+
+    index(input, &[Seq::new(0.0, dims[0] as f64 - 1.0, 1.0)
```
Comment (9prady9) [inline src/index.rs]: Same as above.

**Comment by 9prady9** [comment]
@jramapuram Can also please add slices wrapper, it would be exactly same except that the Seq using the parameters would be along third dimension.

**Comment by jramapuram** [comment]
Cool will refactor to seq default & add span

File: src/index.rs (hunk 1 above)
Comment (jramapuram) [inline src/index.rs]: Will refactor to span.

What about assertion of ranges? I.e. making sure first & last are contained within the max_rows?

Comment (9prady9) [inline src/index.rs]: Should be fine i think, `index` call will return appropriate error code if something is wrong. I don't think we should duplicate error checks in rust wrapper unless absolutely required.

Comment (jramapuram) [inline src/index.rs]: :+1:

**Comment by jramapuram** [comment]
@9prady9 : Refactored and added setters. Please review.

**Comment by 9prady9** [comment]
:+1:

**Pull request merged**